src/main/java/decodes/cwms/CwmsTimeSeriesDAO.java

```diff
@@ -150,1401 +58,1475 @@
+            synchronized(cache)
+            {
+                cache.remove(tsid.getKey());
+            }
+            refreshTsView();
+        }
+        catch(Exception ex)
+        {
+            String msg = "Cannot delete time series '{}'";
+            log.atWarn()
+               .setCause(ex)
+               .log(msg, tsid.getUniqueString());
+        }
+    }
+
+    @Override
+    public CTimeSeries makeTimeSeries(TimeSeriesIdentifier tsid)
+        throws DbIoException, NoSuchObjectException
+    {
+        DbKey sdi = tsid.getKey();
+        CTimeSeries ret = new CTimeSeries(sdi, tsid.getInterval(),
+            tsid.getTableSelector());
+        ret.setTimeSeriesIdentifier(tsid);
+        ret.setDisplayName(tsid.getDisplayName());
+        return ret;
+    }
+
+    @Override
+    public ArrayList<TimeSeriesIdentifier> listTimeSeries()
+        throws DbIoException
+    {
+        // MJM 20161025 don't reload more if already done within threshold.
+        if (System.currentTimeMillis() - lastCacheReload > cacheReloadMS)
+        {
+            reloadTsIdCache();
+        }
+
+        ArrayList<TimeSeriesIdentifier> ret = new ArrayList<TimeSeriesIdentifier>();
+        synchronized(cache)
+        {
+            for (Iterator<TimeSeriesIdentifier> tsidit = cache.iterator(); tsidit.hasNext();)
+            {
+                ret.add(tsidit.next());
+            }
+        }
+        return ret;
+    }
+
+    @Override
+    public ArrayList<TimeSeriesIdentifier> listTimeSeries(boolean forceRefresh)
+        throws DbIoException
+    {
+        if (forceRefresh)
+        {
+            lastCacheReload = 0L;
+        }
+        return listTimeSeries();
+    }
+
+
+    @Override
+    public void reloadTsIdCache() throws DbIoException
+    {
+        log.debug("reloadTsIdCache()");
+
+        // Each TSID will need a site, so prefill the site cache to prevent
+        // it from doing individual reads for each site.
+//        siteDAO.fillCache();
+
+        String q = cwmsTsidQueryBase + " WHERE upper(a.DB_OFFICE_ID) = upper(?)";
+        int origFetchSize = getFetchSize();
+        try
+        {
+            int tsidFetchSize = DecodesSettings.instance().tsidFetchSize;
+            if (tsidFetchSize > 0)
+            {
+                setFetchSize(tsidFetchSize);
+            }
+
+
+            List<TimeSeriesIdentifier> tsidList = getResultsIgnoringNull(q, rs ->
+            {
+                CwmsTsId retVal = null;
+                try
+                {
+                    retVal = rs2TsId(rs, false);
+                }
+                catch (DbIoException | NoSuchObjectException ex)
+                {
+                    log.atWarn()
+                       .setCause(ex)
+                       .log("Error creating Cwms TSID -- skipped.");
+                }
+                return retVal;
+            }, dbOfficeId);
+
+            synchronized(cache)
+            {
+                for(TimeSeriesIdentifier tsid: tsidList)
```

Comment:
Should this cache be cleared before reloading?

Reply:
hmm, yeah.